build_qtkeychain.bat

```diff
@@ -0,0 +1,63 @@
+SETLOCAL
+echo ---- Building qtkeychain ----
+set QTKEYCHAIN_PATH=qtkeychain-0.9.1
+set QT_VERSION=5.11.1
+
+set VALRETURN=0
+
+if %MACHINE_X86% (
+  set PLATFORM=Win32
+  set COMPILERPROJECT="Visual Studio 14 2015"
+) else (
+  set PLATFORM=x64
+  set COMPILERPROJECT="Visual Studio 14 2015 Win64"
+)
+
+if %CONFIG_RELEASE% (
+  set CONFIG=RelWithDebInfo
+) else (
+  set CONFIG=Debug
+)
+
+REM Remove any existing files
+erase %LIB_DIR%\*qtkeychain*
+
+cd build\%QTKEYCHAIN_PATH%
+REM Re-generate solution files.
+rd /S /Q %PLATFORM%
+mkdir %PLATFORM%
+cd %PLATFORM%
+
+REM TODO: verify that the cmake setting BUILD_SHARED_LIBS works correctly.
+SET ADDITIONAL_SETTINGS=
+IF %STATIC_LIBS% (
+SET ADDITIONAL_SETTINGS=-DQTKEYCHAIN_STATIC=ON
+)
+
+    
+"%CMAKEDIR%\cmake" .. -G %COMPILERPROJECT% %ADDITIONAL_SETTINGS% -DBUILD_TEST_APPLICATION=OFF -DCMAKE_PREFIX_PATH=%ROOT_DIR%\Qt-%QT_VERSION%\lib\cmake -DCMAKE_INSTALL_PREFIX=%ROOT_DIR% -DCMAKE_INSTALL_INCLUDEDIR=%INCLUDE_DIR%
+IF ERRORLEVEL 1 (
+    SET VALRETURN=1
+    goto END
+)
+
+"%CMAKEDIR%\cmake" --build . --config %CONFIG% --clean-first 
```

Comment:
I'm worried about this -- can we stick with how the other cmake builds are done and generate a VS solution, then use %MSBUILD% to build it? The risk is that we might build with the wrong platform toolset, or the wrong runtime type. We use the %MSBUILD% environment variable to control this.

Reply:
cmake does this behind the scenes and calls msbuild with the right parameter set. It always uses the right msbuild matching the used compiler. IMHO this is more save than worrying with that ourselves. 
This the way Qt Creator uses cmake by the way. I think we can rely on it.

Reply:
I think we should else consider to change the others as well.

Reply:
Thanks -- I agree now that we resolved my worries in #58. Can you add the `-T %PLATFORM_TOOLSET%` here too?